components/external-services-mock/cmd/main.go

```diff
@@ -59,16 +59,26 @@ func initHTTP(cfg config) http.Handler {
 	router.HandleFunc("/v1/healtz", health.HandleFunc)
 
 	configChangeRouter := router.PathPrefix("/audit-log/v2/configuration-changes").Subrouter()
-	configChangeRouter.Use(authMiddleware)
+	configChangeRouter.Use(oauthMiddleware)
 	configurationchange.InitConfigurationChangeHandler(configChangeRouter, configChangeHandler)
 
 	router.HandleFunc("/audit-log/v2/oauth/token", oauthHandler.Generate).Methods(http.MethodPost)
+	router.HandleFunc("/oauth/token", oauthHandler.Generate).Methods(http.MethodPost)
```

Comment:
Since you're going to reuse the same oathHandler, fix the `APP_AUDITLOG_CLIENT_ID` and `APP_AUDITLOG_CLIENT_SECRET` environment variables to something generic, so as to not imply it's only used only for Auditlog testing.

Also externalize the Basic credentials similarly to that, because there's currently and implicit contract between the external-services-mock implementation and the corresponding e2e test that the basic credentials are "admin" & "admin".